### PR TITLE
Add decomposition for `aten.upsample_bilinear2d.vec`

### PR DESCRIPTION
**Pull request by vivekkhandelwal1**

### CODE REVIEW

**Comment by vivekkhandelwal1** [comment]
Hi @Chillee, I have added the decomposition for `aten.upsample_bilinear2d.vec` op, could you please review this?

**Comment by silvasean** [comment]
Is there a way to write this using a combination of other ops? Right now this reads more like a "loop lowering / reference impl" than a decomposition.

**Comment by Chillee** [comment]
@silvasean yeah that's what I was discussing with @vivekkhandelwal1. I imagine it could be done with some advanced indexing?

Perhaps @ngimel or @jansel have some thoughts.

**Comment by ngimel** [comment]
jax is doing something clever with creating weights matrix and einsum https://jax.readthedocs.io/en/latest/_modules/jax/_src/image/scale.html#scale_and_translate, but it's also possible that loop lowering is the way to go (but in this case it should be in the backend, and not in refs).

**Comment by jansel** [comment]
I think this could be done with indirect indexing.  Rather than doing math in a loop, do to same ops on `torch.arange()` basically computing every element at once.

**Comment by vivekkhandelwal1** [comment]
> -

Could you please elaborate it more? Like how could this be done without loops?

**Comment by jansel** [comment]
```
    for b in range(n_batch):
        for c in range(n_channels):
            for i in range(out_h):
                for j in range(out_w):
```
would become:
```
    b = torch.arange(n_batch, dtype=torch.int64):
    c = torch.arange(n_channels, dtype=torch.int64):
    i = torch.arange(out_h, dtype=torch.int64):
    j = torch.arange(out_w, dtype=torch.int64):
```

max/min/math.floor/math.ceil become torch.amax/torch.amin/torch.floor/torch.ceil and operate on every index at once.

You then just always do the 4 element average (that last case in the if).

**Comment by vivekkhandelwal1** [comment]
Hi @jansel @Chillee, I have modified the code to make use of other torch ops for computations instead of doing scalar computations in the loops as follows: https://github.com/pytorch/pytorch/blob/2be4bc30a6ade5478d3742f4630bde28f332dd29/torch/_decomp/decompositions.py#L1303-L1322

But, I'm not able to figure out a way to index the input tensors via using these tensors https://github.com/pytorch/pytorch/blob/2be4bc30a6ade5478d3742f4630bde28f332dd29/torch/_decomp/decompositions.py#L1303-L1306 since each of them are of different shapes. Could you guys please help me with this part of the logic?

**Comment by jansel** [comment]
I think you only need indirect indexing in the h/w dimensions.  You can do something like:
```
>>> data = torch.randn(1, 1, 100, 100)
>>> x=torch.arange(3, dtype=torch.int64)
>>> y=torch.arange(4, dtype=torch.int64)
>>> data[:,:,x,:][:,:,:,y]
tensor([[[[-1.0041,  2.1739, -0.6829, -1.0448],
          [-1.0356,  1.8366,  1.1109, -0.9281],
          [ 1.4307, -1.7135, -0.9523, -1.2037]]]])
>>> data[:,:,x*3+10,:][:,:,:,y*2+1]
tensor([[[[-1.6775,  1.4742, -0.0412, -1.3619],
          [-0.7262, -0.7640,  0.3876, -0.9727],
          [ 0.6425, -0.5035, -0.3862,  2.3952]]]])
>>> 
```

**Comment by vivekkhandelwal1** [comment]
Hi @jansel, I have updated the code but it's not giving the correct output. The reason is that this code: https://github.com/pytorch/pytorch/blob/68cd09dae5e6666c504b6db0eee7cd7aee09e7ae/torch/_decomp/decompositions.py#L1326-L1333 is not equivalent to this code: https://gist.github.com/vivekkhandelwal1/f0dc90da0c211452472b122c81eb1680#file-bilinear_interpolation-py-L26-L44. 
For example, when this condition is true https://gist.github.com/vivekkhandelwal1/f0dc90da0c211452472b122c81eb1680#file-bilinear_interpolation-py-L26 i.e., the corner case, the current non-loop version evaluates result to 0 since `x_ceil == x_floor == x` makes https://github.com/pytorch/pytorch/blob/68cd09dae5e6666c504b6db0eee7cd7aee09e7ae/torch/_decomp/decompositions.py#L1331-L1333 `q1 == q2 == q == 0`. While in the original implementation the result is directly taken from the input tensor https://gist.github.com/vivekkhandelwal1/f0dc90da0c211452472b122c81eb1680#file-bilinear_interpolation-py-L27. 

Overall, the edge and the corner cases are not handled by the non-loop version. Any help/suggestion/feedback on this would be appreciated.  @Chillee @makslevental

**Comment by jansel** [comment]
Could you map the conditionals to `torch.where()`?
```
torhc.where(
  torch.logical_and(torch.eq(x_ceil, x_floor), torch.eq(y_ceil, y_floor)),
  tensor_if_true,
  tensor_if_false
)
```

**Comment by vivekkhandelwal1** [comment]
Hi @jansel @Chillee, I am finally able to get the non-loop version of this op's decomposition working. Please review the code. I am greatly thankful to you guys for the help.

File: torch/_decomp/decompositions.py [hunk 1]
```diff
@@ -1263,3 +1263,100 @@ def kl_div_backward(
...
+    x_ceil = torch.ceil(x)
+    x_ceil = torch.where(x_ceil > in_h - 1, in_h - 1, x_ceil)
+    y_floor = torch.floor(y)
+    y_ceil = torch.ceil(y)
+    y_ceil = torch.where(y_ceil > in_w - 1, in_w - 1, y_ceil)
+
+    result = torch.zeros(size=(n_batch, n_channels, out_h_int, out_w_int), dtype=input.dtype)
+    result_shape = result.shape
+
+    x_repeat = torch.repeat_interleave(x, out_w_int).view(out_h_int, out_w_int).repeat(n_batch, n_channels, 1, 1).view(result_shape)
```
Comment (jansel) [inline torch/_decomp/decompositions.py:1329]: Why do we need all of these calls to repeat?  I'd expect to be able to handle this with broadcasting.

```
x_view = x.view(1, 1, len(x), 1)
y_view = x.view(1, 1, 1, len(y))
```

**Review by jansel — CHANGES_REQUESTED** [review]

**Comment by jansel** [comment]
If you compare to float64, is the decomp further away from the fp64 "ground truth"?  Or is it just noise?

File: torch/_decomp/decompositions.py [hunk 2]
```diff
@@ -1245,3 +1245,88 @@ def fast_pow(x, ord):
...
+
+    if out_w > 1:
+        if align_corners:
+            w_scale_factor = (in_w - 1) / (int(out_w) - 1)
+        else:
+            w_scale_factor = in_w / out_w
+    else:
+        w_scale_factor = 0.0
+
+    i = torch.arange(out_h)
```
Comment (Chillee) [inline torch/_decomp/decompositions.py:1284]: You should pass the appropriate device (and dtype, but I guess that's int?).

**Review by Chillee — COMMENTED** [review]
A 1e-8 difference is totally fine (for fp32).

You should test on fp16 as well (the existing test should already do this).

I suspect you might need to upcast your input tensor to fp32 before doing the mixing of the pixels, but we'll see.

**Comment by vivekkhandelwal1** [comment]
> A 1e-8 difference is totally fine (for fp32).
> 
> You should test on fp16 as well (the existing test should already do this).
> 
> I suspect you might need to upcast your input tensor to fp32 before doing the mixing of the pixels, but we'll see.

@Chillee Actually, `torch.ops.aten.upsample_bilinear2d.vec` is not implemented for half. While testing with fp16 input it gives the following error:
```
torch.ops.aten.upsample_bilinear2d.vec(inp, output_size=None, align_corners=False, scale_factors=[2.5, 2.0])
  File "/home/vivek/work/02_07/mlir_venv/lib/python3.8/site-packages/torch/_ops.py", line 49, in __call__
    return self._op(*args, **kwargs or {})
RuntimeError: "compute_indices_weights_linear" not implemented for 'Half'
```

**Comment by vivekkhandelwal1** [comment]
> If you compare to float64, is the decomp further away from the fp64 "ground truth"? Or is it just noise?

@jansel I tested with the fp64 input, and the frequency of values with differences increases. The difference is ranging from 1e-6 to 1e-8.

**Comment by vivekkhandelwal1** [comment]
> > If you compare to float64, is the decomp further away from the fp64 "ground truth"? Or is it just noise?
> 
> @jansel I tested with the fp64 input, and the frequency of values with differences increases. The difference is ranging from 1e-6 to 1e-8.

Sorry, I mistakenly gave incompatible types. With the `fp64` input, the decomposition is similar to `ground truth`.

**Comment by jansel** [comment]
Those differences seem small enough not to worry about.  Thanks for checking!

**Comment by ngimel** [comment]
fp16 cpu support isn't expected, but on cuda it should be supported.

**Comment by jansel** [comment]
@Chillee I tried this one out, and it looks like the `arange` call (and everything that depends on it) isn't getting captured in the graph, and they end up becoming constants.

Do we support factory functions in decomps?

**Comment by jansel** [comment]
I rewrote this to not require the `torch.where`.  The trick is in the `1.0 - xscale2` line.
```
@register_decomposition(torch.ops.aten.upsample_bilinear2d.vec)
def upsample_bilinear2d_vec(
    input: Tensor,
    output_size: Optional[List[int]],
    align_corners: bool,
    scale_factors: Optional[List[float]],
) -> Tensor:
    *_, in_h, in_w = input.shape

    if output_size is not None:
        out_h = float(output_size[0])
        out_w = float(output_size[1])
    elif scale_factors is not None:
        out_h = in_h * scale_factors[0]
        out_w = in_w * scale_factors[1]

    # Calculate horizontal and vertical scaling factor
    if out_h > 1:
        if align_corners:
            h_scale_factor = (in_h - 1) / (int(out_h) - 1)
        else:
            h_scale_factor = in_h / out_h
    else:
        h_scale_factor = 0.0

    if out_w > 1:
        if align_corners:
            w_scale_factor = (in_w - 1) / (int(out_w) - 1)
        else:
            w_scale_factor = in_w / out_w
    else:
        w_scale_factor = 0.0

    # FIXME: this is a giant hack to get more of this traced, remove me please!!!
    please_capture_this = torch.zeros_like(input[:1, 0, 0, 0], dtype=torch.int64)
    arange = please_capture_this + torch.arange(max(out_h, out_w), dtype=torch.float32, device=input.device)

    i = arange[:int(out_h)]
    j = arange[:int(out_w)]

    if align_corners:
        x = h_scale_factor * i
        y = w_scale_factor * j
    else:
        x = (h_scale_factor * (i + 0.5) - 0.5).clamp(min=0.0)
        y = (w_scale_factor * (j + 0.5) - 0.5).clamp(min=0.0)

    x_floor = torch.floor(x)
    x_ceil = torch.ceil(x).clamp(max=in_h - 1)
    y_floor = torch.floor(y)
    y_ceil = torch.ceil(y).clamp(max=in_w - 1)

    v1 = input[:, :, x_floor.to(torch.int64), :][:, :, :, y_floor.to(torch.int64)]
    v2 = input[:, :, x_ceil.to(torch.int64), :][:, :, :, y_floor.to(torch.int64)]
    v3 = input[:, :, x_floor.to(torch.int64), :][:, :, :, y_ceil.to(torch.int64)]
    v4 = input[:, :, x_ceil.to(torch.int64), :][:, :, :, y_ceil.to(torch.int64)]

    xscale2 = (x - x_floor).view(-1, 1)
    xscale1 = 1.0 - xscale2

    yscale2 = y - y_floor
    yscale1 = 1.0 - yscale2

    q1 = torch.mul(v1, xscale1) + torch.mul(v2, xscale2)
    q2 = torch.mul(v3, xscale1) + torch.mul(v4, xscale2)
    result = torch.mul(q1, yscale1) + torch.mul(q2, yscale2)
    return result.to(input.dtype)
```

There is still the issue that a `torch.arange` constant shows up in the graph and isn't captured.

**Comment by Chillee** [comment]
> Do we support factory functions in decomps?

Hmm, it should... but we might have an issue with this today. I think the underlying cause might be that we're not enabling re-entrant dispatch today, which was (kind of) a hack to work around inconsistent operator support back in the day. This line of PRs should fix it: https://github.com/pytorch/pytorch/pull/81598

**Comment by vivekkhandelwal1** [comment]
Hi @jansel @Chillee, do I need to make further changes to this PR?

**Comment by ezyang** [comment]
I wanted to shout out here that @zdevito's first class dims will make this substantially easier to write; here is an implementation of grid sampler for example: https://gist.github.com/ezyang/d6fac0b399363c94e8bb0b36b911944f

File: torch/_decomp/decompositions.py [hunk 3]
```diff
@@ -1247,3 +1247,98 @@ def fast_pow(x, ord):
...
+    v4 = input[:, :, x_ceil.to(torch.int64), :][:, :, :, y_ceil.to(torch.int64)]
+    q1 = torch.mul(v1, x_ceil_view - x_view) + torch.mul(v2, x_view - x_floor_view)
+    q2 = torch.mul(v3, x_ceil_view - x_view) + torch.mul(v4, x_view - x_floor_view)
+    result = torch.mul(q1, y_ceil_view - y_view) + torch.mul(q2, y_view - y_floor_view)
+
+    # When (x_ceil == x_floor) and (y_ceil == y_floor).
+    result_cond1 = input[:, :, x.to(torch.int64), :][:, :, :, y.to(torch.int64)]
+
+    # When (x_ceil == x_floor).
+    q1 = input[:, :, x.to(torch.int64), :][:, :, :, y_floor.to(torch.int64)]
```
Comment (jansel) [inline torch/_decomp/decompositions.py:1321]: This is the same value as `v1` (and also `v2`)

File: torch/_decomp/decompositions.py [hunk 4]
```diff
@@ -1247,3 +1247,98 @@ def fast_pow(x, ord):
...
+    q2 = input[:, :, x.to(torch.int64), :][:, :, :, y_ceil.to(torch.int64)]
```
Comment (jansel) [inline torch/_decomp/decompositions.py:1322]: This is the same value as `v3` (and also `v4`)

File: torch/_decomp/decompositions.py [hunk 5]
```diff
@@ -1247,3 +1247,98 @@ def fast_pow(x, ord):
...
+    result_cond2 = torch.mul(q1, y_ceil_view - y_view) + torch.mul(
+        q2, y_view - y_floor_view
+    )
+
+    # When (y_ceil == y_floor).
+    q1 = input[:, :, x_floor.to(torch.int64), :][:, :, :, y.to(torch.int64)]
```
Comment (jansel) [inline torch/_decomp/decompositions.py:1328]: This is the same value as `v1` and `v3`.

File: torch/_decomp/decompositions.py [hunk 6]
```diff
@@ -1247,3 +1247,98 @@ def fast_pow(x, ord):
...
+    q2 = input[:, :, x_ceil.to(torch.int64), :][:, :, :, y.to(torch.int64)]
```
Comment (jansel) [inline torch/_decomp/decompositions.py:1329]: This is the same value as `v2` and `v4`

File: torch/_decomp/decompositions.py [hunk 7]
```diff
@@ -1247,3 +1247,98 @@ def fast_pow(x, ord):
...
+    v1 = input[:, :, x_floor.to(torch.int64), :][:, :, :, y_floor.to(torch.int64)]
+    v2 = input[:, :, x_ceil.to(torch.int64), :][:, :, :, y_floor.to(torch.int64)]
+    v3 = input[:, :, x_floor.to(torch.int64), :][:, :, :, y_ceil.to(torch.int64)]
+    v4 = input[:, :, x_ceil.to(torch.int64), :][:, :, :, y_ceil.to(torch.int64)]
+    q1 = torch.mul(v1, x_ceil_view - x_view) + torch.mul(v2, x_view - x_floor_view)
+    q2 = torch.mul(v3, x_ceil_view - x_view) + torch.mul(v4, x_view - x_floor_view)
+    result = torch.mul(q1, y_ceil_view - y_view) + torch.mul(q2, y_view - y_floor_view)
+
+    # When (x_ceil == x_floor) and (y_ceil == y_floor).
+    result_cond1 = input[:, :, x.to(torch.int64), :][:, :, :, y.to(torch.int64)]
```
Comment (jansel) [inline torch/_decomp/decompositions.py:1318]: In this case `v1`, `v2`, `v3`, and `v4` will all point to the same memory address and hold the same value.

Reading the value again here will generate slower code.

File: torch/_decomp/decompositions.py [hunk 8]
```diff
@@ -1247,3 +1247,98 @@ def fast_pow(x, ord):
...
+    # When (y_ceil == y_floor).
+    q1 = input[:, :, x_floor.to(torch.int64), :][:, :, :, y.to(torch.int64)]
+    q2 = input[:, :, x_ceil.to(torch.int64), :][:, :, :, y.to(torch.int64)]
+    result_cond3 = torch.mul(q1, x_ceil_view - x_view) + torch.mul(
+        q2, x_view - x_floor_view
+    )
+
+    result = torch.where(torch.eq(x_ceil_view, x_floor_view), result_cond2, result)
+    result = torch.where(torch.eq(y_ceil_view, y_floor_view), result_cond3, result)
+    result = torch.where(
```
Comment (jansel) [inline torch/_decomp/decompositions.py:1336]: This can be done without any calls to `where` (see my comment for an example)

**Review by jansel — CHANGES_REQUESTED** [review]

File: torch/_decomp/decompositions.py [hunk 9]
```diff
@@ -1247,3 +1247,98 @@ def fast_pow(x, ord):
...
+
+    x_view = x.view(1, 1, len(x), 1)
+    x_floor_view = x_floor.view(1, 1, len(x_floor), 1)
+    x_ceil_view = x_ceil.view(1, 1, len(x_ceil), 1)
+
+    y_view = y.view(1, 1, 1, len(y))
+    y_floor_view = y_floor.view(1, 1, 1, len(y_floor))
+    y_ceil_view = y_ceil.view(1, 1, 1, len(y_ceil))
+
+    v1 = input[:, :, x_floor.to(torch.int64), :][:, :, :, y_floor.to(torch.int64)]
```
Comment (ngimel) [inline torch/_decomp/decompositions.py:1309]: better avoid double indexing by doing something like
```
x_floor = x_floor.unsqueeze(1).to(int64)
y_floor = y_floor.unsqueeze(0).to(int64)
v1 = input[:, :, x_floor, y_floor]
```
similarly in all other places where you are doing double indexing.

File: torch/_decomp/decompositions.py [hunk 10]
```diff
@@ -1247,3 +1247,98 @@ def fast_pow(x, ord):
...
+        y = (w_scale_factor * (j + 0.5) - 0.5).clamp(min=0.0)
+
+    x_floor = torch.floor(x)
+    x_ceil = torch.minimum(torch.ceil(x), torch.tensor(in_h - 1))
+    y_floor = torch.floor(y)
+    y_ceil = torch.minimum(torch.ceil(y), torch.tensor(in_w - 1))
+
+    x_view = x.view(1, 1, len(x), 1)
+    x_floor_view = x_floor.view(1, 1, len(x_floor), 1)
+    x_ceil_view = x_ceil.view(1, 1, len(x_ceil), 1)
```
Comment (ngimel) [inline torch/_decomp/decompositions.py:1303]: avoid view functions unless necessary, in this case enough to do 
```suggestion
    x_view = x.unsqueeze(-1)
    x_floor_view = x_floor.unsqueeze(-1)
    x_ceil_view = x_ceil.unsqueeze(-1)
```
the rest of the dimensions will be taken care of with implicit broadcast.

File: torch/_decomp/decompositions.py [hunk 11]
```diff
@@ -1247,3 +1247,98 @@ def fast_pow(x, ord):
...
+
+    y_view = y.view(1, 1, 1, len(y))
+    y_floor_view = y_floor.view(1, 1, 1, len(y_floor))
+    y_ceil_view = y_ceil.view(1, 1, 1, len(y_ceil))
```
Comment (ngimel) [inline torch/_decomp/decompositions.py:1307]: views not necessary here, implicit broadcast for ops with y, y_floor, y_ceil is enough.

**Review by jansel — APPROVED** [review]

**Comment by vivekkhandelwal1** [comment]
@jansel could you please merge this PR for me, I don't have merge access for this repo.

**Comment by jansel** [comment]
Thanks @vivekkhandelwal1

If you are looking for something similar, we also need a decomp for `upsample_bilinear2d_backward`.

**Comment by Chillee** [comment]
Awesome! Thanks @vivekkhandelwal1, this was one of the decompositions I was dreading needing to write!

**Review by Chillee — APPROVED** [review]

**Comment by makslevental** [comment]
@ngimel i think this failed to merge last time because of f64 difference 

https://github.com/pytorch/pytorch/runs/7434459402?check_suite_focus=true#step:9:7341

not sure if it's a logic error or what

**Comment by ngimel** [comment]
Ah ok, then tolerances should be adjusted?

**Comment by makslevental** [comment]
> Ah ok, then tolerances should be adjusted?

```
RuntimeError: Difference from float64 is larger with decomposition upsample_bilinear2d.vec than original on output 0. Original max diff: 0.000244140625, Decomp max diff: 0.00048828125
```

I think this is too high for it just to be a noise issue?

**Comment by Chillee** [comment]
Perhaps we need to upcast the input to fp32 at the beginning if it's fp16?

File: torch/_decomp/decompositions.py [hunk 12]
```diff
@@ -1237,3 +1237,73 @@ def fast_pow(x, ord):
...
+
+
+@register_decomposition(torch.ops.aten.upsample_bilinear2d.vec)
+def upsample_bilinear2d_vec(
+    input: Tensor,
+    output_size: Optional[List[int]],
+    align_corners: bool,
+    scale_factors: Optional[List[float]],
+) -> Tensor:
+    # get dimensions of original image
```
Comment (makslevental) [inline torch/_decomp/decompositions.py:1249]:
```suggestion
    input = input.to(dtype=torch.float32) if input.dtype == torch.float16 else input
    
```

Comment (Chillee) [reply torch/_decomp/decompositions.py:1249]: You also need to downcast it at the end.

You could also just try adding a `@pw_cast_for_opmath` decorator.

File: torch/_decomp/decompositions.py [hunk 13]
```diff
@@ -1237,3 +1237,73 @@ def fast_pow(x, ord):
...
+    xscale2 = x_view - x_floor_view
+    xscale1 = 1.0 - xscale2
+
+    yscale2 = y - y_floor
+    yscale1 = 1.0 - yscale2
+
+    q1 = torch.mul(v1, xscale1) + torch.mul(v2, xscale2)
+    q2 = torch.mul(v3, xscale1) + torch.mul(v4, xscale2)
+    result = torch.mul(q1, yscale1) + torch.mul(q2, yscale2)
+    return result.to(input.dtype)
```
Comment (makslevental) [inline torch/_decomp/decompositions.py:1309]:
```suggestion
    return result.to(orig_dtype)
```

**Comment by vivekkhandelwal1** [comment]
> Thanks @vivekkhandelwal1
> 
> If you are looking for something similar, we also need a decomp for `upsample_bilinear2d_backward`.

As of now, we only need this op. If required, I would work on that. Thanks for your help.

**Comment by vivekkhandelwal1** [comment]
> Awesome! Thanks @vivekkhandelwal1, this was one of the decompositions I was dreading needing to write!

Haha, I was equally scared of writing a decomp for this op. Btw, thanks for the help.

**Comment by Chillee** [comment]
For obtaining the backwards pass, one possibility is to just differentiate this decomposition we wrote lol.

**Comment by ezyang** [comment]
@Chillee I think this should be the recommended method

**Comment by ngimel** [comment]
Test failure is still real

**Comment by jansel** [comment]
> @Chillee I think this should be the recommended method

I like this idea.  Is there an easy way to do this automatically?

**Comment by makslevental** [comment]
still failing due to NA https://github.com/pytorch/pytorch/runs/7471640354?check_suite_focus=true#step:9:10301

```
E               Greatest absolute difference: 2.288818359375e-05 at index (0, 3) (up to 1e-05 allowed)
E               Greatest relative difference: 5.165573860818778e-06 at index (0, 3) (up to 1.3e-06 allowed)
```

which is pretty close but others are worse

```
E               AssertionError: Tensor-likes are not close!
E               
E               Mismatched elements: 62 / 576 (10.8%)
E               Greatest absolute difference: 9.918212890625e-05 at index (0, 2, 5, 0, 0) (up to 1e-05 allowed)
E               Greatest relative difference: 0.00024129587364752394 at index (0, 2, 3, 1, 0) (up to 1.3e-06 allowed)
E               
E               The failure occurred for item [1]
test/test_ops.py:1348: AssertionError
```

**Comment by Chillee** [comment]
@makslevental that test looks like functorch flakiness actually.

**Comment by Chillee** [comment]
I think we can just force land? cc: @ngimel @ezyang

**Comment by zou3519** [comment]
Those tests are functorch flakiness, feel free to ignore them

**Comment by makslevental** [comment]
> I think we can just force land? cc: @ngimel @ezyang

image

did i do that right?

**Pull request closed without merging**